lxd/backup.go

```diff
@@ -236,6 +237,11 @@ func backupGetInfo(r io.ReadSeeker) (*backupInfo, error) {
 		return nil, fmt.Errorf("Backup is missing index.yaml")
 	}
 
+	// Check whether the backup has expired
+	if time.Now().Unix()-result.Expiry.Unix() >= 0 {
+		return nil, fmt.Errorf("Backup has expired")
+	}
```

Comment:
So that's storing and complaining about the expiry, but not actually causing its deletion?

Reply:
I don't follow. What's there to delete? When running `lxc export`, a tarball is created which has the `index.yaml` containing the expiry. Then, when running `lxc import`, it checks the expiry date. Nothing is actually created if the backup has expired.

Reply:
The API can be used to create a backup which will not self destruct on download but has an expiry field set. Unless the user manually causes a DELETE, that container will remain on the server's filesystem indefinitely.

Reply:
Similarly, if you run `lxc export` and loose network connection to the server, mid-export, the container will be kept on the server indefinitely (we had a few cases of that).

Reply:
You're absolutely right, I didn't consider this case.

Reply:
So I'm confused as to why this check is even still here.
Expired backups should just go away and when they do, you'll just be getting a 404.

If for some reason we've failed to delete an expired backup, the APIs should behave as if it hasn't expired yet.

Reply:
Ah, I think I understand why there may have been a bit of confusion here.

Backup expiry is purely a server side thing, the expiry itself shouldn't be recorded inside the backup tarball, it should only be kept in the database.